exist-core/src/test/xquery/xquery3/concat.xql

```diff
@@ -87,11 +94,66 @@ function concat:nodes() {
 declare
     %test:assertError("FOTY0013")
 function concat:bad-argument() {
-    ("abc" || "abc" ||  fn:concat#3)
+    ("abc" || "abc" || fn:concat#3)
 };
 
-(: declare
+(: test for #1828 :)
+declare
+    %test:assertEquals("||||")
+function concat:mode-safe-element-content() {
+    <a>||</a> || '||'
+};
+
+declare
+    %test:assertEquals("||||")
+function concat:mode-safe-attribute-value() {
+    <a b="||" />/@b || '||'
+};
+
+declare
+    %test:assertEquals("<a>bc</a>")
+function concat:enclosed-in-element-content() {
+    <a>{ "b" || "c" }</a>
+};
+
+declare
+    %test:assertEquals("<a b='12'/>")
+function concat:decimal-in-attribute-value() {
+    <a b="{ 1 || 2 }"/>
+};
+
+declare
+    %test:assertEquals("<a b='cd'/>")
+function concat:variables-in-attribute-value() {
+    let $c := "c"
+    let $d := "d"
+    return <a b="{ $c || $d }"/>
+};
+
+(:~
+ : The two tests below cannot be commented in.
+ : They cause a parsing error and therefore must
+ : stay like this, until the underlying issue has
+ : been fixed
+ :)
```

Comment:
Is there an issue that has been opened that we can paste a link to here please?

Reply:
@adamretter In response to your suggestion, @line-o has added a link - see the commit https://github.com/eXist-db/exist/pull/3493/commits/88a7fb0c6634dea4f9475c777b95b85141c1a6f0.

Reply:
Yes I saw. I was just waiting for CI.

Reply:
Ah! Great.